### DISCUSSION TITLE
DJL Monthly Update - March 2021

### DISCUSSION THREAD

**Discussion by zachgk** [deepjavalibrary/djl#804] [News]
## Community

@kimim contributed the [DJL Clojure](https://github.com/scicloj/clj-djl) integration into the data and scientific Clojure community [SCICLOJ](https://scicloj.github.io/pages/about/).

@eitansela created a demo of AWS Lambda with Docker: “Serverless TensorFlow Lite Image Classification Model Serving with Deep Java Library” [[Link]](https://github.com/aws-samples/aws-lambda-docker-serverless-inference/tree/main/djl-tensorflow-lite-inference-docker-lambda).

@mgubaidullin created a demo of DJL with Camel-K [[Link]](https://github.com/mgubaidullin/dj-kamel).

@skirdey and Netflix open sourced [Derand](https://github.com/Netflix/derand), a library for high performance text tokenization and cleanup using a character CNN model.

@mgubaidullin upgraded the [Apache Camel DJL Component](https://camel.apache.org/components/latest/djl-component.html) so DJL 0.10.0 will be available with Camel 3.10.0.

### Events

@roywei and @skirdey presented in the St. Louis *Apache Kafka* Meetup on using DJL to deploy models from streaming applications on March 3rd https://www.meetup.com/Saint-Louis-Kafka-meetup-group/events/276466753/?isFirstPublish=true

@goswamig and @skirdey presented on deploying deep learning models using DJL from Scala-based big data pipelines in the Bay Area Big Data and Cloud Meetup group on March 18 https://www.meetup.com/datariders/events/276592801/

@frankliu will introduce DJL to the NY Java Sig meetup on 4/7 
https://us02web.zoom.us/j/86175263218?pwd=aUEwb2drdGZEVmwxcStZOXFibGtKdz09 

@lanking520 will present how DJL works with Rapids on Spark 3.0 to enable running deep learning on GPU from Spark at [NVIDIA’s GTC conference](https://www.nvidia.com/en-us/gtc/) (4/12- 4/16) [[Registration required to view link](https://gtc21.event.nvidia.com/media/Deep%20Learning%20in%20Scala%20on%20Spark%203.0%20with%20GPU%20on%20AWS%20%5BS32285%5D/1_10miapj4)]

## Feature Enhancements

* DJL-Serving homebrew package is online [https://formulae.brew.sh/formula/djl-serving]
* Add setBoolean method [#784](https://github.com/awslabs/djl/pull/784)
* Refactor TensorFlow NDArray [#796](https://github.com/awslabs/djl/pull/796)
* Upgraded TensorFlow native package to 2.4.1 [#747](https://github.com/awslabs/djl/pull/747)
* Upgraded TensorFlow engine to 0.3.0 [#753](https://github.com/awslabs/djl/pull/753)
* Add Training Multi-threaded capabilities [#743](https://github.com/awslabs/djl/pull/743)
* Add String Tensor support for ONNXRuntime [#724](https://github.com/awslabs/djl/commit/a059417104217b22d4fc800e50c0602401f827ee)
* Added Stream loading feature for PyTorch [#729](https://github.com/awslabs/djl/commit/a90129ee518c091f0b83184d3ce470a2f84969d4)
* Added “disable alternative Engine support" for all engines that don’t have operators. [#729](https://github.com/awslabs/djl/commit/a90129ee518c091f0b83184d3ce470a2f84969d4)
* Remove memory scope and improve memory management [#695](https://github.com/awslabs/djl/pull/695)
* Remove erroneous random forest application [#726](https://github.com/awslabs/djl/pull/726)
* Configure verbose for mxnet extra libraries [#728](https://github.com/awslabs/djl/pull/728)
* Fix concat operator on edge case [#739](https://github.com/awslabs/djl/pull/739)
* Add missing softSign test and remove the unused rnn ops [#741](https://github.com/awslabs/djl/pull/741)
* Add ModelView Download URIs for each Model Attribute [#719](https://github.com/awslabs/djl/pull/719)
* Added BlockFactory to DJL for better model packaging[#697](https://github.com/awslabs/djl/pull/697)
* Added String tensor support for ONNXRuntime [#724](https://github.com/awslabs/djl/pull/724)
* Added InputStream loading option for PyTorch [#729](https://github.com/awslabs/djl/pull/729)
* Add MXNet android template project [#146](https://github.com/aws-samples/djl-demo/pull/146)
* Fixed SageMaker SDK for endpointConfig and endPoint deployment [#714](https://github.com/awslabs/djl/pull/714)

## Documentation & Examples

* Add profiler document [#722](https://github.com/awslabs/djl/pull/722)

### D2L Book

* [D2L] Section 8.5 Implementation of Recurrent Neural Networks from Scratch [#88](https://github.com/aws-samples/d2l-java/pull/88)
* [D2L] Section 8.6 Concise Implementation of Recurrent Neural Networks [#89](https://github.com/aws-samples/d2l-java/pull/89)
* [D2L] Section 8.7: Backpropagation Through Time [#90](https://github.com/aws-samples/d2l-java/pull/90)
* [D2L] Section 9.1: Gated Recurrent Units (GRU) [#92](https://github.com/aws-samples/d2l-java/pull/92)
* [D2L] Section 9.2: Long Short-Term Memory [#93](https://github.com/aws-samples/d2l-java/pull/93)

## Bug Fixes

* Fixed the nightly GPU build [#758](https://github.com/awslabs/djl/pull/758)
* Changed build.gradle for serving + nightly build to publish the tar file into djl-ai prod s3 account. [#757](https://github.com/awslabs/djl/pull/757)
* Fixed [ArrayIndex out of range](https://github.com/awslabs/djl/issues/720) [#756](https://github.com/awslabs/djl/pull/756)
* Added Code coverage for PyTorch and TensorFlow packages

**Comment by hmf** [comment]
Would be nice if the presentations are recorded for later viewing.
If possible could you also make available the date and location of the events? For example the Zoom above lets me know the date and time, but I could not figure out my local time from that.

Thank you.

**Reply by zachgk** [reply]
That's a good suggestion. We'll try to ask about recordings of presentations and make sure all the recordings show up in the next monthly update

Here are links with some details on the NY Java SIG Presentation: https://www.javasig.com/ and https://www.eventbrite.com/e/deep-learning-using-deep-java-library-tickets-148702979615. It is at Wed, April 7, 2021 3:30 PM – 5:30 PM PDT.
